### DISCUSSION TITLE
[Idea]: audio description for blind and visually impaired.

### DISCUSSION THREAD

**Discussion by rickinhissmart** [sircharlo/meeting-media-manager#3112] [Ideas] [answered]
### Is your idea related to a problem?

feature enhancement

### Describe your idea

I hope this message finds you well. I am writing to propose a feature enhancement for your media display application that could significantly benefit blind and visually impaired users.
As you may know, audio descriptions are crucial for providing context and enhancing the experience for individuals who are unable to see the visuals. However, synchronizing audio descriptions with the meeting can be quite challenging, especially in a group setting where multiple devices are in use.
I would like to suggest the implementation of a web server feature that displays the filename and timestamp of media currently being played. This would enable the development of an application that connects to devices on the same Wi-Fi network, allowing users to download the media and play it simultaneously on their personal devices.
Additionally, your application could offer an option to download audio description versions of the media and stream them directly to users’ devices via Wi-Fi. or a streaming service like icecast server , This would create a seamless experience where users can listen to the content with headphones, ensuring they can fully enjoy the experience without interruption.
I believe that these enhancements would greatly improve accessibility and inclusivity for blind and visually impaired individuals, allowing them to engage more fully with the jw.org valuable resources.
Thank you for considering this suggestion. I would be happy to discuss this further or provide any additional information that may be helpful.

https://www.jw.org/en/library/videos/#en/categories/VODMidweekMeetingAD

Kind regards Rick,

### Describe alternatives you've considered

none

**Comment by mtdvlpr** [comment]
Hi @rickinhissmart, thanks for your input!

To be clear. What do you expect from M3 precisely? Making the filename and timestamp of currently playing media somehow available for third-parties?

**Reply by sircharlo** [reply]
I think he means adding a web server feature that would stream synced audio from the AD version of whatever media is playing on the monitors.

**Reply by rickinhissmart** [reply]
trying to find a solution , this feature could lead to this being used in Kingdom Halls, a server would be a good idea

**Comment by sircharlo** [comment]
While I can definitely see the advantages of a solution like the one you describe, it's a very niche situation. I think for now we'll focus on implementing and stabilizing features that can benefit a greater number of users, but maybe we can come back to this in the future if it's still a need.

If anyone wants to contribute a PR to implement this, it would of course be greatly appreciated.

**Comment by rickinhissmart** [comment]
could we just have a simple webpage that displays a list of downloaded files
 and a simple text that syncs timestamp of video playing with the current file name, then could build a sync app for that?

**Reply by sircharlo** [reply, marked as answer]
Hi @rickinhissmart,  

Thank you for your thoughtful suggestions to improve accessibility for blind and visually impaired individuals. We appreciate your enthusiasm and ideas of how such features could be implemented and used.  

However, please understand that the features you're describing—like a web server capable of displaying currently playing media and timestamps, or a media streaming service allowing multiple device connections—are significant enhancements that would introduce substantial complexity to the app.  

Currently, these features fall outside the project's scope. That said, contributions to lay the groundwork for these ideas are always welcome. For example:  
- A basic web server feature displaying the currently playing media's filename, URL, and timestamp.  
- A streaming server that synchronizes and serves the AD version of the media.  

Each of these features would need to be implemented step-by-step as individual components before achieving the functionality you're envisioning. If you'd like to contribute a pull request, we'd be happy to review it.  

We appreciate your understanding and support for making the app more inclusive.  

Kind regards,  
The M³ Team
